File: skywalker/src/read_joints.py
#!/usr/bin/env python
import rospy
import pandas as pd
import numpy as np
import scipy.io as sc
from sensor_msgs.msg import JointState
import sys
import copy
import rospy
import moveit_commander
import moveit_msgs.msg
import geometry_msgs.msg
from math import pi
from std_msgs.msg import String
from moveit_commander.conversions import pose_to_list
import csv
joint0 = [1,2]
joint1= [1,2]
joint2= [1,2]
joint3 = [1,2]
joint4= [1,2]
joint5= [1,2]
joint6 = [1,2]
poses  = []


class MoveGroupPythonIntefaceTutorial(object):
  """MoveGroupPythonIntefaceTutorial"""
  def __init__(self):
    super(MoveGroupPythonIntefaceTutorial, self).__init__()

    ## BEGIN_SUB_TUTORIAL setup
    ##
    ## First initialize `moveit_commander`_ and a `rospy`_ node:
    moveit_commander.roscpp_initialize(sys.argv)
    rospy.init_node('move_group_python_interface_tutorial',
                    anonymous=True,  disable_signals=True)

    ## Instantiate a `RobotCommander`_ object. This object is the outer-level interface to
    ## the robot:
    robot = moveit_commander.RobotCommander()

    ## Instantiate a `PlanningSceneInterface`_ object.  This object is an interface
    ## to the world surrounding the robot:
    scene = moveit_commander.PlanningSceneInterface()

    ## Instantiate a `MoveGroupCommander`_ object.  This object is an interface
    ## to one group of joints.  In this case the group is the joints in the Panda
    ## arm so we set ``group_name = panda_arm``. If you are using a different robot,
    ## you should change this value to the name of your robot arm planning group.
    ## This interface can be used to plan and execute motions on the Panda:
    group_name = "manipulator"
    group = moveit_commander.MoveGroupCommander(group_name)
    ## END_SUB_TUTORIAL

    ## BEGIN_SUB_TUTORIAL basic_info
    ##
    ## Getting Basic Information
    ## ^^^^^^^^^^^^^^^^^^^^^^^^^
    # We can get the name of the reference frame for this robot:
    planning_frame = group.get_planning_frame()
    print "============ Reference frame: %s" % planning_frame

    # We can also print the name of the end-effector link for this group:
    eef_link = group.get_end_effector_link()
    print "============ End effector: %s" % eef_link

    # We can get a list of all the groups in the robot:
    group_names = robot.get_group_names()
    print "============ Robot Groups:", robot.get_group_names()

    # Sometimes for debugging it is useful to print the entire state of the
    # robot:
    print "============ Printing robot state"
    print robot.get_current_state()
  def execute_plan(self, plan):
    # Copy class variables to local variables to make the web tutorials more clear.
    # In practice, you should use the class variables directly unless you have a good
    # reason not to.
    group = self.group
    group.execute(plan, wait=True)

  def callback(self,n):

    # joint0.append(data.position[0])
    # joint1.append(data.position[1])
    # joint2.append(data.position[2])
    # joint3.append(data.position[3])
    # joint4.append(data.position[4])
    # joint5.append(data.position[5])
    # joint6.append(data.position[6])
    # print (data.position[0])
    x = group.get_current_pose().pose
    writer.writerow([x.position.x, x.position.y, x.position.z, x.orientation.x,
      x.orientation.y, x.orientation.z, x.orientation.w])
    print "I am in the call back"
if __name__ == '__main__':    
  tutorial = MoveGroupPythonIntefaceTutorial()
  counter =10
  f = open('/home/usman/usman-ros/src/skywalker/skywalker/src/joint_positions.csv', 'w')
  writer = csv.writer(f, delimiter=',', quotechar='"', quoting=csv.QUOTE_MINIMAL)
  try:
    group_name = "manipulator"
    group = moveit_commander.MoveGroupCommander(group_name)
    rospy.Subscriber("/joint_states", JointState, tutorial.callback)
    print "============ Press `Enter` to begin the tutorial by setting up the moveit_commander (press ctrl-d to exit) ..."
    rospy.spin()
      
  except KeyboardInterrupt:
    print "The program has been terminated!"
    f.close()


# print ('here I am')
# nx = np.array((joint0,joint1,joint2,joint3,joint4,joint5,joint6))
# adict = {}
# adict['nx'] = nx
# sc.savemat('file.mat',adict)
# pd.DataFrame({'joint0': joint0, 'joint1': joint1, 'joint2':joint2, 'joint3':joint3, 'joint4':joint4, 'joint5':joint5, 'joint6':joint6}).to_csv('file.csv', index=False)

File: skywalker/src/read_joints.cpp


/* Author: Sachin Chitta, Dave Coleman, Mike Lautman */

#include <moveit/move_group_interface/move_group_interface.h>
#include <moveit/planning_scene_interface/planning_scene_interface.h>

#include <moveit_msgs/DisplayRobotState.h>
#include <moveit_msgs/DisplayTrajectory.h>

#include <moveit_msgs/AttachedCollisionObject.h>
#include <moveit_msgs/CollisionObject.h>

int main(int argc, char** argv)
{
  ros::init(argc, argv, "move_group_interface_tutorial");
  ros::NodeHandle node_handle;
  ros::AsyncSpinner spinner(1);
  spinner.start();

  static const std::string PLANNING_GROUP = "manipulator";

  // The :move_group_interface:`MoveGroup` class can be easily
  // setup using just the name of the planning group you would like to control and plan for.
  moveit::planning_interface::MoveGroupInterface move_group(PLANNING_GROUP);
  moveit::planning_interface::MoveGroupInterface::Plan my_plan;

  // We will use the :planning_scene_interface:`PlanningSceneInterface`
  // class to add and remove collision objects in our "virtual world" scene
  moveit::planning_interface::PlanningSceneInterface planning_scene_interface;

  // Raw pointers are frequently used to refer to the planning group for improved performance.
  const robot_state::JointModelGroup* joint_model_group =
  move_group.getCurrentState()->getJointModelGroup(PLANNING_GROUP);
  move_group.clearPathConstraints();

  geometry_msgs::Pose target_pose3 = move_group.getCurrentPose().pose;

  //geometry_msgs::Pose target_pose3 = move_group.getCurrentPose("tool0").pose;
  std::cout << "The end effector link is "<<move_group.getEndEffectorLink() << std::endl;
  std::cout << "The planning frame is "<< move_group.getPlanningFrame() << std::endl;

  std::cout << target_pose3.position.x<<", "<< target_pose3.position.y<<", "<< target_pose3.position.z<<", "<< target_pose3.orientation.x<<", "
  << target_pose3.orientation.y<<", "<< target_pose3.orientation.z<<", "<< target_pose3.orientation.w<<", "<<std::endl;

  ros::shutdown();
  return 0;
}
